libraries/freertos_plus/standard/https/include/iot_https_client.h

```diff
@@ -0,0 +1,387 @@
+ * that this function does not fail.
+ * 
+ * Header lines are appended in the following format:
+ * 
+ * "header_field_name: header_value\r\n"
+ * 
+ * Example:
+ * 
+ * "Range: bytes=1024-2047\r\n"
+ * 
+ * The last header line requires a "\r\n" following to separate the last header line from the entity body. 
+ * These 2 characters are accounted for in #requestUserBufferMinimumSize.
+ * 
+ * The remaining length, after the header is added, is printed to the system configured standard debug output when 
+ * IOT_LOG_LEVEL_HTTPS is set to IOT_LOG_DEBUG in iot_config.h. 
+ * 
+ * For a synchronous request, if extra headers are desired to be added, this function must be invoked before 
+ * @ref https_client_function_sendsync.
+ * <b> Synchronous Example </b>
+ * @code{c}
+ * ...
+ * char * date_in_iso8601[17];
+ * GET_DATE_IN_ISO8601(date_in_iso8601);
+ * IotHttpsClient_AddHeader(reqHandle, "x-amz-date", date_in_iso8601, strlen(date_in_iso8601));
+ * ...
+ * IotHttpsClient_SendSync(&connHandle, reqHandle, &respHandle);
+ * ...
+ * @endcode
+ * 
+ * The following header fields are automatically added to the request header buffer and should be NOT added again with 
+ * this routine:
+ * - Connection:
+ * - User-agent:
+ * - Host:
+ * - Content-Length:
+ *
+ * @param[in] reqHandle - HTTPS request to write the header line to.
+ * @param[in] pName - String header field name to write. This is NULL terminated.
+ * @param[in] pValue - https header value buffer pointer. Do not include token name.
+ * @param[in] len - length of header value to write.
+ * 
+ * @return #IOT_HTTPS_OK if the header line was successfully added to req.request_message.headers.
+ *         #IOT_HTTPS_INSUFFICIENT_MEMORY if the header line cannot fit into the header buffer.
+ *         #IOT_HTTPS_INVALID_PARAMETER for NULL parameters or if an attempt to add automatic headers is done.
+ *         Please see #IotHttpsReturnCode_t for other failure codes.
+ */
+/* @[declare_https_client_addheader] */
+IotHttpsReturnCode_t IotHttpsClient_AddHeader(IotHttpsRequestHandle_t reqHandle, char * pName, char * pValue, uint32_t len);
+/* @[declare_https_client_addheader] */
+
+/**
+ * @brief Synchronous send of the the HTTPS request.
+ * 
+ * This function blocks waiting for the entirety of sending the request and receiving the response. 
+ * 
+ * #IotHttpsRequestInfo_t.pSyncInfo must have been configured to send a request synchronously. If the
+ * request was configured with #IotHttpsRequestInfo.pAsyncInfo instead, invoking this function will have 
+ * undefined behavior. 
+ * 
+ * If parameter pConnHandle is passed in pointing to a NULL #IotHttpsConnectionHandle_t, then a connection is to be made 
+ * implicitly and a valid connection handle in pConnHandle will be returned. 
+ * If parameter pConnHandle points to a non-NULL #IotHttpsConnectionHandle_t and it is in a disconnected state, then a 
+ * connection is to be made implicitly and valid connection handle in pConnHandle will be returned. 
```

Comment:
TCP keep-alive messages are automatic in the TCP stack and not controlled by the HTTP library. 
Just to re-iterate, "persistent" connection only means that the server/client does not close the connection immediately after the response to the request with the "Connection: keep-alive" header. 

We cannot mark automatically that the connection is in a disconnected state. This can only happen in IotHttpsClient_Disconnect(). A network error will be returned to the client when the server closed the connection.